projects/plugins/jetpack/extensions/blocks/button/button.php

```diff
@@ -226,7 +220,7 @@ function get_button_wrapper_styles( $attributes ) {
 	$has_width = array_key_exists( 'width', $attributes );
 
 	if ( $has_width ) {
-		$styles[] = 'max-width: 100%';
+		$styles[] = sprintf( 'width: %s;', $attributes['width'] );
```

Comment:
Apply the width to the root of the `jetpack/button`, instead of its `__link` element that's now 100% wide.